File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        mx, curMx = -inf, 0
        
        for num in nums:                           
                                                    
            curMx += num                            
            mx = max(mx, curMx)
            curMx = max(0, curMx)
            
        return mx

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        vector<int> dp(nums.size());
        dp[0]=nums[0];
        for(int i=1; i<nums.size(); i++){
            dp[i]=max(nums[i], nums[i]+dp[i-1]);
        }
        return *max_element(dp.begin(), dp.end());
    }
};